packages/react/src/components/TopNavigation/TopNavigation.tsx

```diff
@@ -0,0 +1,224 @@
+import clsx from 'clsx';
+import {Box, Container, IconButton, Toolbar, Typography, useMediaQuery} from '@mui/material';
+import {useColorScheme, useTheme, Theme} from '@mui/material/styles';
+import {ChevronDownIcon, HamburgerIcon} from '@oxygen-ui/react-icons';
+import {Mode} from '@mui/system/cssVars/useCurrentColorScheme';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+import './top-navigation.scss';
+import Avatar from '../Avatar';
+import Button, {ButtonProps} from '../Button';
+import Link from '../Link';
+import UserDropdownMenu, {ModeListInterface} from '../UserDropdownMenu';
+import AppBar, {AppBarProps} from '../AppBar';
+
+/**
+ * Interface for the Top Navigation component props.
+ */
+export interface TopNavigationProps extends AppBarProps {
+  /**
+   * Brand information.
+   */
+  brand?: BrandTemplate;
+  /**
+   * Is the left navigation bar activated.
+   */
+  isLeftNavigationActive?: boolean;
+  /**
+   * Links available on the top navigation bar.
+   */
+  links?: Omit<ButtonProps, 'color'>[];
+  /**
+   * List of modes.
+   */
+  modes?: ModeListInterface[];
+  /**
+   * Function to handle left navigation bar button toggle.
+   */
+  onLeftNavigationTrigger?: () => void;
+  /**
+   * Function on user logging out.
+   */
+  onLogOut?: () => void;
+  /**
+   * Logged user information.
+   */
+  user?: UserTemplate;
+}
+
+/**
+ * Interface for the brand template.
+ */
+export interface BrandTemplate {
+  logo?: {
+    desktop?: ReactNode;
+    mobile?: ReactNode;
+  };
+  onClick?: () => void;
+  title?: ReactNode;
+}
+
+/**
+ * Interface for the logged user template.
+ */
+export interface UserTemplate {
+  email?: string;
+  image?: string;
+  name?: string;
+}
+
+const COMPONENT_NAME: string = 'TopNavigation';
+
+/**
+ * Top Navigation component.
+ */
+const TopNavigation: FC<TopNavigationProps> & WithWrapperProps = (props: TopNavigationProps): ReactElement => {
+  const {
+    className,
+    children,
+    isLeftNavigationActive,
+    brand,
+    user,
+    links,
+    modes,
+    onLogOut,
+    onLeftNavigationTrigger,
+    ...rest
+  } = props;
+
+  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
+  const theme: Theme = useTheme();
+  const {mode, setMode} = useColorScheme();
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const classes: string = clsx(
+    'oxygen-top-nav',
+    {
+      mobile: isMobile,
+    },
+    className,
+  );
+
+  const open: boolean = Boolean(anchorElUser);
+
+  const handleOpenUserMenu = (event: MouseEvent<HTMLElement>): void => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = (): void => {
+    setAnchorElUser(null);
+  };
+
+  const onModeChange = (selectedMode: Mode): void => {
+    setMode(selectedMode);
+  };
+
+  return (
+    <AppBar position="static" color="inherit" variant="outlined" elevation={0} className={classes} {...rest}>
```

Comment:
We can add a role here for screen reader a11y
Ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role

```suggestion
    <AppBar role="navigation" position="static" color="inherit" variant="outlined" elevation={0} className={classes} {...rest}>
```

Reply:
ack. role added.